analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SecurityPInvokeMethodShouldNotBeCalled.cs

```diff
@@ -53,6 +53,10 @@ class Program
         public static void CoInitializeSecurity(int param) { } // Compliant non extern
         public extern void CoInitializeSecurity(string param); // Compliant non static
         public static extern int CoInitializeSecurity(int param1, string param2); // Compliant no DllImport
+        [DllImport(null)] // Error [CS0591]
```

Comment:
another compile error case for safety would be 
```
[DllImport]
```

Reply:
to clarify what I wanted with the second case

for 
```
LibraryImport(libraryName:"x")]
```
the AST is 
![image](https://user-images.githubusercontent.com/38876598/206471445-7113f133-61da-49cf-928a-86b38964eb56.png)

for 
```
[System.Runtime.InteropServices.LibraryImport(libraryName:"foo", EntryPoint="bar")]
```
the AST is for the `EntryPoint` property is ([sharplab](https://sharplab.io/#v2:C4LghgzsA+ACBMBGAsAKFgZgAQKwYSwG80tSsBtWRAOgCUBXAO2AEsBbAU2oElmOAnAPYAHAMoCAbiwDGHCNQAyLAEb8w/AJ7c2wwf2AAKADYq1mgHJhOIAEQAzQYJsAaLAFFmmgAqCWzALw2yuo2AJQAuiRkmDiIAGxYHAAewAKMOAAsWACyBqFYANxoAL5AA==))
![image](https://user-images.githubusercontent.com/38876598/206472244-7c536ff0-927e-4a7f-8e15-1098f05bcfb8.png)

Both the cases above compile.

For the compilation error case:
```
[System.Runtime.InteropServices.LibraryImport(EntryPoint="bar")]
```
there is no argumentList, it's just the parameter. This does not compile, however ArgumentList is `null`. This ensures safety in the unlikely case that the API will make the parameter optional, or will allow to specify it via a property.

Reply:
Hm now I might be wrong 🤔 

I remember I hit this problem in the past where there is no argument, but only properties in the attribute.

Reply:
The info is taken from the semantic model and there is a distinction between constructor and named arguments: https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.attributedata?view=roslyn-dotnet-4.3.0